e2e/testcases/helm_sync_test.go

```diff
@@ -205,47 +178,56 @@ service:
 		}
 	  }`)
 
-	err := nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion("15.2.35")),
-		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "wordpress"}))
-	if err != nil {
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must reference valid ConfigMaps in spec.helm.valuesFileRefs: ConfigMap \"helm-watch-config-map\" not found")
+
+	cmName := "helm-watch-config-map"
+	nt.T.Log("Apply valid ConfigMap that is not immutable (which should not be allowed)")
+	cm0 := fake.ConfigMapObject(core.Name(cmName), core.Namespace(configsync.ControllerNamespace))
+	cm0.Immutable = pointer.Bool(false)
+	cm0.Data = map[string]string{"something-else.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+`,
+	}
+	nt.T.Cleanup(func() {
+		if err := nt.KubeClient.Delete(cm0); err != nil {
+			nt.T.Log(err)
+		}
+	})
+	if err := nt.KubeClient.Create(cm0); err != nil {
 		nt.T.Fatal(err)
 	}
-	var expectedCPURequest string
-	var expectedCPULimit string
-	var expectedMemoryRequest string
-	var expectedMemoryLimit string
-	if nt.IsGKEAutopilot {
-		expectedCPURequest = "250m"
-		expectedCPULimit = expectedCPURequest
-		expectedMemoryRequest = "512Mi"
-		expectedMemoryLimit = expectedMemoryRequest
-	} else {
-		expectedCPURequest = "150m"
-		expectedCPULimit = "1"
-		expectedMemoryRequest = "250Mi"
-		expectedMemoryLimit = "300Mi"
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must reference valid ConfigMaps in spec.helm.valuesFileRefs; ConfigMap \"helm-watch-config-map\" in namespace \"config-management-system\" is not immutable")
```

Comment:
got a semi colon that should be a colon here. I've noticed this in a few of your errors in the past. The standard go pattern is to join error messages with colons. The error on the left provides more context about the error on the right, recursively.

That said, the configsync K- errors are kind of rule breakers, since they start with a capital letter. I'm not sure if there's some other convention I don't know about for semicolon use.

Reply:
I changed these all to colons. I think you're right that we should try to follow the standard go pattern here where we can.